app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -371,7 +394,8 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
                     errorResponse: WebResourceResponse?
                 ) {
                     Log.e(TAG, "onReceivedHttpError: ${errorResponse?.statusCode} : ${errorResponse?.reasonPhrase} for: ${request?.url}")
-                    if (request?.url.toString() == loadedUrl) {
+                    if (request?.isForMainFrame == true && request.url.toString() == loadedUrl) {
+                        pageError = errorResponse?.statusCode ?: 404
```

Comment:
Why fallback to 404? HTTP errors can be for various codes; if unknown is a valid response the `pageError` variable should be nullable.